### PR TITLE
Candidate for 1.1.1.post1

### PR DESCRIPTION
**Pull request by dgpv**
#44

### CODE REVIEW

File: bitcointx/core/psbt.py [hunk 1]
```diff
@@ -2133,6 +2141,13 @@ def stream_serialize(self, f: ByteStream_Type,
         for outp in self.outputs:
             outp.stream_serialize(f, **kwargs)
 
+    @no_bool_use_as_property
+    def is_final(self) -> bool:
+        for txin_index, _ in enumerate(self.unsigned_tx.vin):
```
Comment (dgpv) [inline bitcointx/core/psbt.py:2146]: The length of `self.inputs` could be larger than len(self.unsigned_tx.vin) if programmer added more inputs, but did not update unsigned_tx. But if the PSBT is consistent re `unsigned_tx.vin`/`inputs`, then this loop should give the correct result.

**Comment by dgpv** [comment]
Also includes fix for #47

**Comment by AdamISZ** [comment]
Did I miss some subtletly, but wouldn't it make sense if the new `PartiallySignedTransaction.is_final()` were used here:

https://github.com/Simplexum/python-bitcointx/blob/b7b15fb4f42d3d3b2abc1b4d6d5803137d8cfcd8/bitcointx/core/psbt.py#L2177

?

**Comment by dgpv** [comment]
calling `PSBT.is_final()` from `PSBT.sign()` It would repeat the loop over the inputs unnecessarily. The loop in `sign()` counts other stats, and it makes sense to count "final" stats along the way and just check with one `==`

**Pull request merged**